### PR TITLE
Plans on site domain update domain purchase card design

### LINKED ISSUE #19830: [Android] Plans on site domain: Update to latest designs
See Figma rvVgXupiyONUcJMgPSoMFj-fi-3512%3A19304

### PR DESCRIPTION
**Pull request by ravishanker**
Fixes #19830 

Update site domain screen items to latest designs

| Site domain | Claim domain | Dark mode
|--------|--------|--------|
||||

-----

## To Test:

- Launch JP app
- `Select` a site with a paid plan, no custom domain, and have admin role
- `Go` to Domains (My site -> domains or My site -> More... -> domain)
- `Verify` a Claim your domain card is displayed as in second image above
- `Tap` on Search for domain
- `Verify` it launches domain purchase flow as before
- `Try` same flow with a site that has custom domain
- `Verify` it looks like the first image above
- `Try` same flow with a site on free plan
- `Verify it displays a **Upgrade to a plan card**

-----

## Regression Notes

1. Potential unintended areas of impact

    - None

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Manual tests

3. What automated tests I added (or what prevented me from doing so)

    - Unit tests

-----

## PR Submission Checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

-----

## UI Changes Testing Checklist:

- [x] Portrait and landscape orientations.
- [x] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

File: WordPress/src/main/res/drawable/img_illustration_domains_card_header.xml
Comment (irfano) [inline WordPress/src/main/res/drawable/img_illustration_domains_card_header.xml:1]: There is also a reference to `img_illustration_domains_card_header.xml` in `baseline.xml`. Let's remove it from there as well.

**Comment by irfano** [comment]
The dark mode looks like this. The "Primary domain" label doesn't look good. @osullivanchris, can we get your feedback on the color of the label?

File: WordPress/src/main/res/layout/domain_purchase_card.xml [hunk 1]
```diff
@@ -39,35 +29,38 @@
...
+            android:id="@+id/search_domains_button"
+            style="@style/DomainsPlanPurchaseCardPrimaryButton"
             android:layout_width="match_parent"
             android:layout_height="wrap_content"
-            android:layout_marginEnd="@dimen/margin_extra_large"
-            android:layout_marginStart="@dimen/margin_extra_large"
-            android:layout_marginTop="@dimen/margin_extra_large"
-            android:text="@string/domains_free_plan_get_your_domain_title"
-            android:textAlignment="center"
+            android:layout_margin="@dimen/margin_extra_large"
```
Comment (irfano) [inline WordPress/src/main/res/layout/domain_purchase_card.xml:58]: - According to the design, there is no bottom, start, end margins on the search button. We should remove ` android:layout_margin`.
- According to the design, there is only top margin, `16dp`.  Also, `MaterialButton` has internal `6dp` padding.  So, we should add `10dp` top margin to the search button to make the total gap between the button and the text `16dp`.


Screenshot 2023-12-21 at 19 15 57

**Comment by ravishanker** [comment]
> The dark mode looks like this. The "Primary domain" label doesn't look good. @osullivanchris, can we get your feedback on the color of the label?
> 

Updated the color for dark mode.  Please see image in PR description 🔝

**Comment by osullivanchris** [comment]
Hey there's two paths we were considering here. One was to adopt the new "All domains" page UI, like this
image

The fallback was to keep the current UI for the site domains screen, adding the card. With some tweaks.

image

Which direction are we taking? This still looks like the fallback safer version where we keep the current UI for this screen, but I see you've changed the 'primary domain' chip per my 'ideal' design adopting the all domains styles.

**Comment by irfano** [comment]
> > The dark mode looks like this. The "Primary domain" label doesn't look good. @osullivanchris, can we get your feedback on the color of the label?
> 
> Updated the color for dark mode. Please see image in PR description 🔝

Looks better! 👍🏻

**Comment by irfano** [comment]
The string for the "Primary domain" label on the "Claim domain" section was incorrect (it was showing "Primary site address" instead). I have updated the screenshot from the latest commit.

**Comment by irfano** [comment]
> Which direction are we taking?

This PR is primarily implements the "fallback" option. 

> This still looks like the fallback safer version where we keep the current UI for this screen, but I see you've changed the 'primary domain' chip per my 'ideal' design adopting the all domains styles.

Thank you for the clarification. I will continue the development of the "ideal" design in other PRs. This PR includes changes from the "ideal" design, but I will merge this one as nothing is broken.

**Review by irfano — APPROVED** [review]

File: WordPress/src/main/res/layout/domain_purchase_card.xml (hunk 1 above)
Comment (irfano) [reply WordPress/src/main/res/layout/domain_purchase_card.xml:58]: This has been resolved with https://github.com/wordpress-mobile/WordPress-Android/pull/19865

File: WordPress/src/main/res/drawable/img_illustration_domains_card_header.xml
Comment (irfano) [reply WordPress/src/main/res/drawable/img_illustration_domains_card_header.xml:1]: Resolved with https://github.com/wordpress-mobile/WordPress-Android/pull/19865.

**Pull request merged**